recipe-server/normandy/base/tests/__init__.py

```diff
@@ -7,20 +9,42 @@
 
 
 class Whatever(object):
-    def __init__(self, test=lambda x: True):
+    def __init__(self, test=lambda x: True, name='unnamed'):
         self.test = test
+        self.name = name
+
+    @classmethod
+    def startswith(cls, prefix):
+        return cls(lambda s: s.startswith(prefix), name=f'startswith {prefix}')
 
     @classmethod
     def endswith(cls, suffix):
-        return cls(lambda s: s.endswith(suffix))
+        return cls(lambda s: s.endswith(suffix), name=f'endswith {suffix}')
 
     @classmethod
     def contains(cls, *values):
-        return cls(lambda s: all(value in s for value in values))
+        name_values = ''.join(str(v) for v in values)
+        return cls(lambda s: all(value in s for value in values), name=f'contains {name_values}')
+
+    @classmethod
+    def iso8601(cls):
+        def is_iso8601_date(s):
+            # Will throw is it does not match
```

Comment:
Rather than throw, this should return false for the comparison. Otherwise things like deep object diffs in assertions won't work with this.